core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -0,0 +1,159 @@
+public class SSHAction extends Action {
+  private final SSHActionConfig config;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SSHAction.class);
+
+  public SSHAction(SSHActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(final ActionContext context) throws Exception {
+    Connection connection = new Connection(config.host);
+    try {
+      connection.connect();
+
+      String password = Strings.isNullOrEmpty(config.password) ? null : config.password;
+      if (!connection.authenticateWithPublicKey(config.user, new File(config.privateKeyFile), password)) {
+        throw new IOException(String.format("Unable to establish SSH connection for %s@%s on port %d",
+                                            config.user, config.host, config.port));
+      }
+
+      LOG.info(String.format("Connection established with the host %s", config.host));
+
+      Session session = connection.openSession();
+      session.execCommand(config.cmd);
+
+      InputStream stdout = new StreamGobbler(session.getStdout());
+      BufferedReader outBuffer = new BufferedReader(new InputStreamReader(stdout));
+      InputStream stderr = new StreamGobbler(session.getStderr());
+      BufferedReader errBuffer = new BufferedReader(new InputStreamReader(stderr));
+
+      StringBuilder outBuilder = new StringBuilder();
+      StringBuilder errBuilder = new StringBuilder();
+
+      String line = outBuffer.readLine();
+      while (line != null) {
+        outBuilder.append(line).append("\n");
+        line = outBuffer.readLine();
+      }
+
+      line = errBuffer.readLine();
+      while (line != null) {
+        errBuilder.append(line).append("\n");
+        line = errBuffer.readLine();
+      }
+
+      LOG.info("Output: {}", outBuilder.toString());
+      LOG.info("Errors: {}", errBuilder.toString());
+
+      if (config.output != null) {
+        context.getArguments().set(config.output, outBuilder.toString());
+      }
+      session.close();
+    } finally {
+      connection.close();
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    this.config.validate();
+  }
+
+  /**
+   * Config class that contains all the properties needed to SSH into the remote machine and run the script.
+   */
+  public static class SSHActionConfig extends PluginConfig {
+
+    @Description("Host name of the remote machine where the command need to be executed.")
+    private String host;
+
+    @Nullable
+    @Description("Port to connect to. Defaults to 22")
+    private int port;
+
+    @Description("User name used for conn")
+    private String user;
+
+    @Description("Private key file.")
+    private String privateKeyFile;
+
+    @Description("Password")
+    private String password;
+
+    @Description("Command to be executed on the remote host.")
+    private String cmd;
+
+    @Nullable
+    @Description("Script output variable")
```

Comment:
ok so its not writing to the secure store, its setting a pipeline argument. And the output of the script means stdout right? Please add some more detail to the description, otherwise users will not know what this is for.